src/device/usbd.c

```diff
@@ -1355,12 +1379,21 @@ void usbd_edpt_close(uint8_t rhport, uint8_t ep_addr) {
   return;
 }
 
-void usbd_sof_enable(uint8_t rhport, bool en) {
+void usbd_sof_enable(uint8_t rhport, sof_consumer_t consumer, bool en) {
   rhport = _usbd_rhport;
 
-  // TODO: Check needed if all drivers including the user sof_cb does not need an active SOF ISR any more.
-  // Only if all drivers switched off SOF calls the SOF interrupt may be disabled
-  dcd_sof_enable(rhport, en);
+  uint8_t consumer_old = _usbd_dev.sof_consumer;
+  // Keep track how many class instances need the SOF interrupt
+  if (en) {
+    _usbd_dev.sof_consumer |= (uint8_t)(1 << consumer);
+  } else {
+    _usbd_dev.sof_consumer &= (uint8_t)(~(1 << consumer));
+  }
+
+  // Test logically unequal
```

Comment:
Seems like a typo, should be: `// Test logically XOR`

Reply:
Not a typo, it could be other consumer later, should be only called when whole value change to true or false.

Reply:
I can see where you coming from. _Keep in mind that my comment has nothing to do with the usage and is only related to the code readability._

While it's being the same and because you are actually using the unequally operator in the expression for logical states, I don't really see an additional value in this comment. Meaning if I didn't knew what the expression would do the comment wouldn't help me at all. Providing a different view, like to mention that this is basically an expression which is equivalent to the XOR gate would. It's the same if this would be used instead: `((!_usbd_dev.sof_consumer && consumer_old) || (_usbd_dev.sof_consumer && !consumer_old))` A different implementation for the same logic where adding a comment like "// Either one but not both" would also not add anything.